firstvoices/backend/views/custom_order_recalculate_views.py

```diff
@@ -1,51 +1,154 @@
+        },
+    ),
+)
+class CustomOrderRecalculateViewset(
+    CustomOrderFVPermissionViewSetMixin,
+    SiteContentViewSetMixin,
+    ListViewOnlyModelViewSet,
 ):
     http_method_names = ["get", "post"]
-    serializer_class = CustomOrderRecalculationPreviewResultDetailSerializer
+    serializer_class = CustomOrderRecalculationResultSerializer
 
     def get_queryset(self):
         site = self.get_validated_site()
         if site.count() > 0:
-            return CustomOrderRecalculationPreviewResult.objects.filter(
-                site__slug=site[0].slug
+            return CustomOrderRecalculationResult.objects.filter(
+                site__slug=site[0].slug, is_preview=False
             ).order_by("-latest_recalculation_date")
         else:
-            return CustomOrderRecalculationPreviewResult.objects.none()
+            return CustomOrderRecalculationResult.objects.none()
 
     def create(self, request, *args, **kwargs):
         site = self.get_validated_site()
         site_slug = site[0].slug
 
+        # Call the recalculation task
+        try:
+            recalculation_results = recalculate_custom_order.apply_async((site_slug,))
+
+            # Delete any previous results
+            CustomOrderRecalculationResult.objects.filter(
+                site=site[0], is_preview=False
+            ).delete()
+
+            # Save the result to the database
+            CustomOrderRecalculationResult.objects.create(
+                site=site[0],
+                latest_recalculation_result=recalculation_results.get(timeout=360),
+                task_id=recalculation_results.task_id,
+                is_preview=False,
+            )
+
+            return Response({"message": "Recalculation has been queued."}, status=201)
+
+        except recalculate_custom_order_preview.OperationalError:
+            raise CeleryError()
+
+    @action(methods=["get"], detail=False, url_path="preview", url_name="preview")
+    def get_preview(self, request, *args, **kwargs):
+        # Get queryset
+        site = self.get_validated_site()
+        if site.count() > 0:
+            queryset = CustomOrderRecalculationResult.objects.filter(
+                site__slug=site[0].slug, is_preview=True
+            ).order_by("-latest_recalculation_date")
+        else:
+            queryset = CustomOrderRecalculationResult.objects.none()
+
+        # List queryset the same way as list() method
+        queryset = utils.filter_by_viewable(request.user, queryset)
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        # non-paginated response
+        serializer = self.serializer_class(
+            queryset, many=True, context={"request": request}
+        )
+        return Response(serializer.data)
+
+    @get_preview.mapping.post
+    @action(detail=False, methods=["post"])
```

Comment:
This decorator creates a duplicate endpoint `api/1.0/sites/test-site/dictionary-cleanup/create_preview/` which only takes a POST request and acts the same as `dictionary-cleanup/preview/`. The decorator is required for the method to function as a non-detail endpoint. If we wanted to get rid of the duplicate endpoint I could potentially combine `get_preview` and `create_preview` into the same method with different actions depending on the request type. I'd like opinions on whether or not to go in this direction